File: src/main/java/com/example/day04/leetcode_24/Solution.java
package com.example.day04.leetcode_24;

/**
 * 两两交换链表的节点
 */
public class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head == null || head.next == null){
            return head;
        }
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode cur = dummy;
        ListNode tmp = null;
        ListNode first, second;
        while(cur.next!= null && cur.next.next != null){
            tmp = cur.next.next.next;
            first = cur.next;
            second = cur.next.next;
            cur.next = second;
            second.next = first;
            first.next = tmp;
            cur = first;
        }
        return dummy.next;
    }
}

File: src/main/java/com/example/day04/leetcode_24/Solution.py
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def swapPairs(self,head: Optional[ListNode]) -> Optional[ListNode]:
       dummy = ListNode(0,head)
       cur = dummy
       while cur.next and cur.next.next:
           first = cur.next
           second = cur.next.next

           first.next = second.next
           second.next = first
           cur.next = second
           cur = first
       return dummy.next# 尤其要注意缩进问题